backend/apps/monkvision/apis/lib/nlp.js

```diff
@@ -0,0 +1,176 @@
+        durationIntent = await getDurationIntent(sentence),
+        resourceIntent = utils.resourceExists(sentence).length ? await getResourceIntent(sentence) : "null";
+
+    return {
+        metric: merticsIntent, duration: durationIntent.weight,
+        durationUnit: durationIntent.unit, resourceName: resourceIntent
+    }
+}
+
+/**
+ * Predict Metrics Intent
+ * @param {string} query - Plain english sentence
+ * @param {Object} metrics - Push mactched intents from sentence
+ * @returns - returns metrics intent
+ */
+async function getMetricsIntent(query, metrics = []) {
+    const encodedQuery = encodeData([{ text: query }]).then((data) => { return data }),
+        metricPredictions = await MODELS["metrics"].predict(await encodedQuery).array(),
+        maxMetricPredict = METRICS[metricPredictions[0].indexOf(Math.max(...metricPredictions[0]))];
+    
+    if (!utils.intentExists(query, maxMetricPredict)) return metrics;
+    query = query.replace(maxMetricPredict, "");
+    metrics.push(maxMetricPredict);
+    return await getMetricsIntent(query, metrics);
+}
+
+/**
+ * Predict Durations Intent
+ * @param {string} query - Plain english sentence
+ * @param {Object} durations - Push mactched intents from sentence
+ * @returns - returns durations intent
+ */
+async function getDurationIntent(query, durations = {}) {
+    const encodedQuery = encodeData([{ text: query }]).then((data) => { return data }),
+        durationPredictions = await MODELS["durations"].predict(await encodedQuery).array(),
+        maxDurationPredict = DURATIONS[durationPredictions[0].indexOf(Math.max(...durationPredictions[0]))],
+        strMatch = utils.getDurationUnit(query, maxDurationPredict);
+
+    if (strMatch.length == 0) {
+        if (Object.keys(durations).length > 1) return utils.calculateDurationInMinutes(durations);
+        durations.unit = Object.keys(durations)[0];
+        durations.weight = Object.values(durations)[0];
+        return utils.calculateDurationInMinutes(durations);
+    }
+    const durationWeight = utils.getDurationWeight(query, maxDurationPredict);
+    if (durationWeight.startWord) {
+        const substr = query.substring(query.indexOf(durationWeight.startWord), query.indexOf(strMatch[0]));
+        query = query.replace(substr + strMatch[0], "");
+        durations[maxDurationPredict] = durationWeight.base + durationWeight.digits;
+    } else {
+        query = query.replace(durationWeight + " " + strMatch[0], "");
+        durations[maxDurationPredict] = durationWeight;
+    }
+    return await getDurationIntent(query, durations);
+}
+
+/**
+ * Predict Resources Intent
+ * @param {string} query - Plain english sentence
+ * @returns returns resources intent
+ */
+async function getResourceIntent(query) {
+    const encodedQuery = encodeData([{ text: query }]).then((data) => { return data }),
+        resourcePredictions = await MODELS["resources"].predict(await encodedQuery).array();
```

Comment:
Would you ever put await before a variable??? What the heck is this code - `await encodedQuery`???